chapter-05/final/money/currency.go

```diff
@@ -0,0 +1,63 @@
+package money
+
+import "strings"
+
+// Currency defines the code of a money and its decimal precision.
+type Currency struct {
+	code      string
+	precision int
+	toEuro    float32
+}
+
+func (c Currency) Code() string {
+	return c.code
+}
+
+// errUnknownCurrency is returned when a currency is unsupported.
+const errUnknownCurrency = moneyError("unknown currency")
+
+// currencies defines the supported currencies.
+var currencies = map[string]Currency{
```

Comment:
Do we want to hide this behind a func? To prevent the possibility of changing the contents, and to use up a little less RAM (but I'm thinking service and not CLI again here)